### PR TITLE
Feature/cpp mock scenario ament cmake

### PR DESCRIPTION
**Pull request by hakuturu583**
## Types of PR

- [x] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

## Description

Enable run cpp_mock_test with colcon test. (with -DWITH_INTEGRATION_TEST=ON)

## How to review this PR.

Please check passing all test cases.

### CODE REVIEW

File: common/simple_junit/include/simple_junit/junit5.hpp [hunk 1]
```diff
@@ -181,6 +181,8 @@ struct SimpleTestCase
 {
   Skipped skipped;
 
+  std::vector<Pass> pass;
```
Comment (yamacir-kit) [inline common/simple_junit/include/simple_junit/junit5.hpp:184]: Please remove this unused and non-standard element `std::vector<Pass> pass`. 
Type `Pass` is not a valid JUnit 5 element.

File: common/simple_junit/include/simple_junit/junit5.hpp [hunk 2]
```diff
@@ -334,10 +355,24 @@ struct SimpleTestSuites : private std::unordered_map<std::string, SimpleTestSuit
     if (not testsuites.name.empty()) {
       current_node.append_attribute("name") = testsuites.name.c_str();
     }
-
+    size_t tests = 0;
+    size_t failures = 0;
+    size_t pass = 0;
+    size_t errors = 0;
```
Comment (yamacir-kit) [inline common/simple_junit/include/simple_junit/junit5.hpp:361]: Please replace `size_t` to `std::size_t`.

**Review by yamacir-kit — CHANGES_REQUESTED** [review]

File: common/simple_junit/include/simple_junit/junit5.hpp [hunk 3]
```diff
@@ -287,10 +298,20 @@ struct SimpleTestSuite : private std::unordered_map<xs::string, SimpleTestCase>
 
     current_node.append_attribute("name") = testsuite.name.c_str();
 
+    size_t tests = 0;
+    size_t failures = 0;
+    size_t pass = 0;
+    size_t errors = 0;
```
Comment (yamacir-kit) [inline common/simple_junit/include/simple_junit/junit5.hpp:304]: `size_t` => `std::size_t`

**Review by yamacir-kit — CHANGES_REQUESTED** [review]

File: common/simple_junit/include/simple_junit/junit5.hpp (hunk 3 above)
Comment (hakuturu583) [reply common/simple_junit/include/simple_junit/junit5.hpp]: Fixed in https://github.com/tier4/scenario_simulator_v2/pull/529/commits/53fee2d8a5fb29abe5efb9a352b175f8a2885e20

**Review by yamacir-kit — APPROVED** [review]

**Pull request merged**